_posts/2017-04-05-promoting-rust.md

```diff
@@ -0,0 +1,27 @@
+---
+layout: post
+title: "Advocating X effectively, where X is Rust"
+author: Florian Gilcher
+description: "Want to convince people of Rust? Here's a few hints!"
+---
+
+Excited about Rust? Want to talk about it all the time? Then stay a while and listen!
+
+A lot of programming language advocacy falls flat by not being well-prepared, placed wrong or just being plain contra-productive. This is wasted effort on all sides: for the writer to write and for the reader to read - and dismiss.
+
+I hope this article is also useful for people advocating other languages or projects.
+
+## Do things that work
+
+If you really want to advocate for a thing, do things that work. This isn't about you to feel better or others feeling worse. This is about convincing others that we are onto something.
+
+
+### What doesn't work
+
+Finally, here's a few things that you definitely shouldn't do, if you have any interest of convincing someone. It's good to always have these in your head, just to ensure to not do them accidentally.
+
+First of all, don't ask random people to consider Rust or an _existing_ project. They already chose the language they want to use for implementation and as long as they don't come up with a specific problem that brings up the language discussion on again, the topic is through. No one wants to waste more time on topics already discussed. This is a little bit different if you are already part of a project, but even there, take care to waste project time and make the discussion fruitful.
+
+Second, don't kick people when they are down. The next security incident in libinternet? This is not the place for drive-by advocacy. Sometimes, this is called [Ambulance Chasing](https://en.wikipedia.org/wiki/Ambulance_chasing). "Rewrite in Rust" just isn't a practical suggestion for existing projects with no Rust programmers around. I'll get to that in a minute.
+
+Third, don't make the impression you take project maintainers for amateurs for their choice of technology. While there's certain things many people would like to go away, it's not like people thought about their choice of technology for building something. The modern Tor probably shouldn't be written in C, but Tor was born at times where it was a good solution.
```

Comment:
Calling Tor out specifically here feels like it may unnecessarily start a flame-war argument.  We could just mention "applications which are security-sensitive have better options for programming languages, but many were written before these options were available"